core/src/main/scala/epollcat/internal/ch/SocketHelpers.scala

```diff
@@ -98,18 +106,26 @@ private[ch] object SocketHelpers {
   }
 
   def getLocalAddress(fd: CInt): SocketAddress = {
-    val addr = stackalloc[posix.netinet.in.sockaddr_in]()
+    val sz =
+      if (preferIPv4Stack)
```

Comment:
From reading the `getsockname` man page, I believe
that you can unconditionally stackalloc a sockaddr_in6
with appropriate addrlen. 

Yes, in the IPv4 case it over-allocates a few (10?) bytes
on the stack, & stack is a precious resource.  If does
simplify the code flow and probably speeds things up
by removing a branch.   

If only that level of detail were our biggest concern.
It is more the reduction in code complexity.

Reply:
Ah that does simplify things quite a bit. I will do that.